app/components/ProductList/productDetail.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styles from './productdetail.css'
+import {Router, Route, Link, params, IndexRoute, hashHistory} from 'react-router'
+
+
+class ProductDetail extends React.Component {  
+
+
+  render() {
+
+    const productsData = this.props.route.data;
+    const id = this.props.params.id;
+
+    const item = productsData.filter(product => {
```

Comment:
Use productsData.find(product => product.id === id)  to get only a single item. Then you also don't need to access it via item[0].